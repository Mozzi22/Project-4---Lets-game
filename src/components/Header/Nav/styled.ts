import styled from 'styled-components';
import { colors } from 'src/components/UI/themeStyles';

export const StNav = styled.div`
    display: flex;
    justify-content: space-evenly;
    align-items: center;
    width: 20%;
        h1{
            font-family: 'Poppins', sans-serif;
            font-size: 25px;
            color: ${({ colors, theme }) => colors[theme].colorTextForm};
        }
        img{
            height: 40px;
        }
    cursor: pointer;
`;
