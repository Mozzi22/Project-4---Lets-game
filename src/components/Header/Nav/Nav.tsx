import React from 'react';
import { useTranslation } from 'react-i18next';
import { useTheme } from 'src/components/Hooks/useTheme';
import Link from 'src/components/UI/Link';
import history from 'src/helpers/history';
import { StNav } from './styled';

const Nav = () => {
    const { t } = useTranslation();
    const { colors, theme } = useTheme();

    const goToMain = () => {
        history.push('/main');
    };
    const goToStats = () => {
        history.push('/statistics');
    };

    return (
        <StNav
            theme={theme}
            colors={colors}
        >
            <img src="/public/assets/images/game-logo.svg" onClick={goToMain} />
            <Link
                theme={theme}
                colors={colors}
                id='main-page-btn'
                content={t('main_page')}
                width='110px'
                fontSize='18px'
                backgroundSize='0'
                padding='0'
                margin='0'
                bgColor='transparent'
                onClick={goToMain}
            />
            <Link
                theme={theme}
                colors={colors}
                id='statistics-btn'
                content={t('statistics')}
                width="110px"
                fontSize='18px'
                backgroundSize="0"
                padding="0"
                margin="0"
                bgColor="transparent"
                onClick={goToStats}
            />
        </StNav>
    );
};

export default Nav;
