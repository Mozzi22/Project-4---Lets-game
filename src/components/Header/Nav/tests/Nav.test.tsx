import React from 'react';
import history from 'src/helpers/history';
import { shallow } from 'enzyme';
import Nav from '../Nav';

jest.mock('src/helpers/history', () => ({
  push: jest.fn(),
}));
jest.mock('react-i18next', () => ({
   useTranslation: jest
     .fn()
     .mockReturnValue({ i18n: { changeLanguage: jest.fn() }, t: jest.fn() }),
 }));

describe('Nav', () => {
   it('Should match snapshot', () => {
    const component = shallow(<Nav />);
    expect(component.html()).toMatchSnapshot();
  });
  it('should call Nav main', () => {
    const component = shallow(<Nav />);
    component
      .find('Link')
      .at(0)
      .getElement()
      .props.onClick();
    expect(history.push).toHaveBeenCalledWith('/main');
  });
   it('should call Nav statistics', () => {
    const component = shallow(<Nav />);
    component
      .find('Link')
      .at(1)
      .getElement()
      .props.onClick();
    expect(history.push).toHaveBeenCalledWith('/statistics');
  });
  it('should render img', () => {
    const component = shallow(<Nav />);
    expect(component.find('img')).toHaveLength(1);
  });
});
