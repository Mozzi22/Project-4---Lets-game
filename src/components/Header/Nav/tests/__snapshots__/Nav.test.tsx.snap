// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Should match snapshot 1`] = `"<div class=\\"styled__StNav-sc-ssmv4t-0 bMovpb\\"><img src=\\"/public/assets/images/game-logo.svg\\"/><button id=\\"main-page-btn\\" type=\\"button\\" font-size=\\"18px\\" width=\\"110px\\" class=\\"styled__StLink-sc-dq9w1g-0 AcUgc\\"></button><button id=\\"statistics-btn\\" type=\\"button\\" font-size=\\"18px\\" width=\\"110px\\" class=\\"styled__StLink-sc-dq9w1g-0 AcUgc\\"></button></div>"`;
